src/Microsoft.DotNet.Build.CloudTestTasks/RunnerScripts/xunitrunner-perf/csvjsonconvertor.py

```diff
@@ -0,0 +1,200 @@
+    jobType.jobTypeName = opts['--jobTypeName']
+    job.jobType = jobType
+
+    jobGroup = serialobj.JobGroup()
+    jobGroup.jobGroupName = opts['--jobGroupName']
+    job.jobGroup = jobGroup
+
+    job.jobDescription = opts['--jobDescription']
+    job.jobName = opts['--jobName']
+
+    root = serialobj.Root()
+    root.job = job
+    jsonOutput = serialobj.JsonOutput()
+    jsonOutput.roots.append(root)
+    with open(opts['--jsonFile'], 'w+') as opfile:
+        opfile.write(jsonOutput.to_JSON())
+        opfile.flush()
+        opfile.close()
+
+    log.info('Conversion of csv to json successful')
+
+def check_required_args(optdict):
+    if not '--csvFile' in optdict or not '--jobName' in optdict or not '--configName' in optdict or not '--userAlias' in optdict or not '--branch' in optdict or not '--osInfoName' in optdict or not '--jsonFile' in optdict:
+        raise getopt.GetoptError
+
+def validate_args(args):
+    # validate required params
+    opts, args = getopt.getopt(args, '', ['csvFile=', 'jsonFile=', 'jobName=', 'jobDescription=', 'configName=', 'jobGroupName=', 'jobTypeName=', 'username=', 'userAlias=', 'branch=', 'buildInfoName=', 'buildNumber=', 'machinepoolName=', 'machinepoolDescription=', 'architectureName=', 'manufacturerName=' , 'microarchName=', 'numberOfCores=', 'numberOfLogicalProcessors=', 'totalPhysicalMemory=', 'osInfoName=', 'osVersion=', 'machineName=', 'machineDescription='])
+    optdict = dict(opts)
+    check_required_args(optdict)
+    return optdict
+
+def main():
+    """
+        Usage::
+            csv_to_json.py
+                --csvFile csvFile to convert to json "MyLoc\\csvfile.csv"
+                --jsonFile output json file location "MyLoc\\jsonfile.json"
+                --jobName "sample job"
+                --jobDescription sample job description
+                --configName "sample config"
+                --jobGroupName sample job group
+                --jobTypeName "Private"
+                --username "sample user"
+                --userAlias "sampleuser"
+                --branch "ProjectK"
+                --buildInfoName "sample build"
+                --buildNumber "1234"
+                --machinepoolName "HP Z210 Workstation"
+                --machinepoolDescription "Intel64 Family 6 Model 42 Stepping 7"
+                --architectureName "AMD64"
+                --manufacturerName "Intel"
+                --microarchName "SSE2"
+                --numberOfCores "4"
+                --numberOfLogicalProcessors "8"
+                --totalPhysicalMemory "16342"
+                --osInfoName "Microsoft Windows 8.1 Pro"
+                --osVersion "6.3.9600"
+                --machineName "PCNAME"
+                --machineDescription "Intel(R) Core(TM) i7-2600 CPU @ 3.40GHz"
+
+
+    """
+
+    opts = validate_args(sys.argv[1:])
+    csvdict = read_csv(opts['--csvFile'])
+    generate_json(opts, csvdict)
+    return 0
+
+if __name__ == '__main__':
+    sys.exit(main())
```

Comment:
Did you consider using the cmdline module? Like my previous comment suggested? It will save and upload the logs from this file, which, if this is being run inside of helix, will make it a lot easier to debug problems.